ts/src/bitmex.ts

```diff
@@ -1857,14 +1860,37 @@ export default class bitmex extends Exchange {
             'ordType': orderType,
             'text': brokerId,
         };
-        if ((orderType === 'Stop') || (orderType === 'StopLimit') || (orderType === 'MarketIfTouched') || (orderType === 'LimitIfTouched')) {
-            const stopPrice = this.safeNumber2 (params, 'stopPx', 'stopPrice');
-            if (stopPrice === undefined) {
-                throw new ArgumentsRequired (this.id + ' createOrder() requires a stopPx or stopPrice parameter for the ' + orderType + ' order type');
-            } else {
-                request['stopPx'] = parseFloat (this.priceToPrecision (symbol, stopPrice));
-                params = this.omit (params, [ 'stopPx', 'stopPrice' ]);
+        const customTriggerType = (orderType === 'Stop') || (orderType === 'StopLimit') || (orderType === 'MarketIfTouched') || (orderType === 'LimitIfTouched');
+        // support for unified trigger format
+        const triggerPrice = this.safeNumberN (params, [ 'triggerPrice', 'stopPx', 'stopPrice' ]);
+        if ((triggerPrice !== undefined) && !customTriggerType) {
+            request['stopPx'] = parseFloat (this.priceToPrecision (symbol, triggerPrice));
+            const triggerDirection = this.safeString (params, 'triggerDirection');
+            params = this.omit (params, [ 'triggerPrice', 'stopPrice', 'stopPx', 'triggerDirection' ]);
+            const triggerAbove = (triggerDirection === 'above');
+            this.checkRequiredArgument ('createOrder', triggerDirection, 'triggerDirection', [ 'above', 'below' ]);
+            this.checkRequiredArgument ('createOrder', side, 'side', [ 'buy', 'sell' ]);
+            if (type === 'limit') {
+                request['price'] = parseFloat (this.priceToPrecision (symbol, price));
+                if (side === 'buy') {
+                    request['ordType'] = triggerAbove ? 'StopLimit' : 'LimitIfTouched';
+                } else {
+                    request['ordType'] = triggerAbove ? 'LimitIfTouched' : 'StopLimit';
+                }
+            } else if (type === 'market') {
+                if (side === 'buy') {
+                    request['ordType'] = triggerAbove ? 'Stop' : 'MarketIfTouched';
+                } else {
+                    request['ordType'] = triggerAbove ? 'MarketIfTouched' : 'Stop';
+                }
+            }
+        } if (customTriggerType) {
+            if (triggerPrice === undefined) {
+                // if exchange specific trigger types were provided
+                throw new ArgumentsRequired (this.id + ' createOrder() requires a triggerPrice (stopPx|stopPrice) parameter for the ' + orderType + ' order type');
```

Comment:
@ttodua this should fix the issue I was talking about

Reply:
ahh, right.